code/game/objects/items/cigs_lighters.dm

```diff
@@ -321,9 +321,38 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 		return
 	var/to_smoke = smoke_all ? (reagents.total_volume * (dragtime / smoketime)) : REAGENTS_METABOLISM
 	var/mob/living/carbon/smoker = loc
-	if(!istype(smoker) || src != smoker.wear_mask)
-		reagents.remove_any(to_smoke)
-		return
+	// These checks are a bit messy but at least they're fairly readable
+	// Check if the smoker is a mob
+	if(!istype(smoker))
```

Comment:
this actually does `!istype(smoker, /mob/living/carbon)` because you pathed it that way on line 323. Is that intentional? I think you should change 323 to be a typecast to `/mob` or `/mob/living` and then do something like `ismob()` or `isliving()`.

actually, line 323 is probably meant to be a carbon based on the whole accesses to `wear_mask`. I think this code needs some re-evaluation based on that.

Reply:
323 is supposed to be a carbon, yeah. I'm guessing the confusion is from my comment. I'll update it